saleor/graphql/product/mutations/product_variant/product_variant_create.py

```diff
@@ -297,6 +300,12 @@ def check_for_duplicates_in_stocks(cls, stocks_data):
     @classmethod
     def save(cls, info: ResolveInfo, instance, cleaned_input):
         new_variant = instance.pk is None
+        instance.track_inventory = (
+            get_track_inventory_by_default()
+            if cleaned_input.get("track_inventory") is None
+            else cleaned_input.get("track_inventory")
+        )
+        instance.save()
```

Comment:
There are two things we need to take care of here.

1. `ProductVariantUpdate` mutation inherits from `ProductVariantCreate`, so this logic would run there as well (which we don't want). One option I see here is to create a method `set_track_inventory` in both mutations and provide different logic. In `ProductVariantCreate` we would have:

```
def set_track_inventory(cls, _info, instance, cleaned_input):
    instance.track_inventory = (
        get_track_inventory_by_default()
        if cleaned_input.get("track_inventory") is None
        else cleaned_input.get("track_inventory")
    )
```

In `ProductVariantUpdate`:
```
def set_track_inventory(cls, _info, instance, cleaned_input):
    track_inventory = cleaned_input.get("track_inventory")
    if track_inventory is not None:
        instance.track_inventory = track_inventory
```

Then in the `save` method, we could only call `set_track_inventory` and in both mutations, the proper implementation of this method would be called.

2. `instance.save()` in line 308 is not needed  as there is a save in line 310.